Backend/core/data_sources/udp_source.py

```
@@ -0,0 +1,145 @@
+        
+        self._running = True
+        self._thread = threading.Thread(target=self._listen_loop, daemon=True)
+        self._thread.start()
+    
+    def _listen_loop(self):
+        """Main listening loop for UDP data."""
+        while self._running:
+            try:
+                readable, _, _ = select.select([self.socket], [], [], 5)
+                if self.socket in readable:
+                    data, addr = self.socket.recvfrom(1024)
```

Ruff (0.14.13)

Unpacked variable `addr` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)